src/content/docs/apis/nerdgraph/examples/nerdgraph-log-data-obfuscation.mdx

```diff
@@ -0,0 +1,561 @@
+---
+title: "NerdGraph tutorial: Data obfuscation for your logs"
+tags:
+  - APIs
+  - NerdGraph
+  - Examples
+  - Logs
+metaDescription: "Create, query, and manage obfuscation rules for your log data with NerdGraph, the New Relic GraphQL explorer."
+---
+
+You control what customer data is logged with our log management service, so be sure to follow your organization's security guidelines to mask, obfuscate, or prevent sending any sensitive data. Our log management service automatically masks number patterns that appear to be for items such as credit cards or Social Security numbers. For more information, see our [security documentation](/docs/logs/get-started/new-relics-log-management-security-privacy) for log management.
+
+We also provide options for you to manage and sanitize parts of your customer data when it arrives at New Relic. This makes it easier and safer for you, rather than needing to track down sensitive data in all of your applications and try to fix them at the source.
+
+You can use NerdGraph at [api.newrelic.com/graphiql](https://api.newrelic.com/graphiql) to create, query, and manage your data obfuscation rules for logs. NerdGraph is our GraphQL-format API explorer.
+
+**Reviewer:** What about the ability to set up obfuscation rules in the UI? I saw a front end design doc in Atlassian. Is there an ETA for that? Will both the API and the UI options be released at the same time?
+
+## Obfuscation and hashing [#definitions]
+
+Obfuscation is when we entirely mask or remove data with no means for retrieval. For example, if a log line `Sent 123-45-6789` becomes `Sent XXX-XX-XXXX`, that's obfuscation.
+
+Hashing is when a sensitive value is replaced with that value, sent through a one-way hashing algorithm. This allows lookup of related records if you have the original value and are able to hash it. For instance, if a log line `Sent 123-45-6789` is hashed to `Sent XYZAB`, the `XYZAB` is the calculated result of hashing `123-45-6789`.
+
+* Field-level obfuscation or hashing applies to the entire attribute on an event.
+* In-message obfuscation or hashing means the rules can target part of a message, such as the `Sent` message examples in this section.
```

Comment:
Do we need to mention this? The users will be defining their own obfuscation expressions, so they will be the ones in charge of specifying whether to hash/mask the whole attribute, or to just hash/mask a part of it. Probably we don't need to mention this, then?

Reply:
agreed. thanks for pointing this out!